src/test/java/echoserver/EchoServerTest.java

```diff
@@ -28,12 +29,14 @@ void returnsInputOfFourDifferentClients() throws IOException, InterruptedExcepti
         echoServer.listenForClients();
         Thread.sleep(100);
 
-        var outputString = output.toString();
-
-        assertThat(outputString).contains(expectedOutput1);
-        assertThat(outputString).contains(expectedOutput2);
-        assertThat(outputString).contains(expectedOutput3);
-        assertThat(outputString).contains(expectedOutput4);
+        assertThat(getFakeClientSocketOutputString(0, fakeServerSocket)).contains(expectedOutput1);
+        assertThat(getFakeClientSocketOutputString(1, fakeServerSocket)).contains(expectedOutput2);
+        assertThat(getFakeClientSocketOutputString(2, fakeServerSocket)).contains(expectedOutput3);
+        assertThat(getFakeClientSocketOutputString(3, fakeServerSocket)).contains(expectedOutput4);
         assertEquals(expectedConnectedClients, echoServer.getClientSockets().size());
     }
+
+    String getFakeClientSocketOutputString(int index, FakeServerSocket fakeServerSocket) {
```

Comment:
Could this be a method on the `FakeServerSocket`?

Reply:
Good point, I'll change this. Thanks